backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -1554,36 +1554,51 @@ export class ExperimentService {
   ): Promise<ValidatedExperimentError[]> {
     logger.info({ message: `Validating experiments` });
 
-    const validationErrors = await Promise.all(
+    const validationErrors = await Promise.allSettled(
       experimentFiles.map(async (experimentFile) => {
         let experiment = JSON.parse(experimentFile.fileContent);
-        const fileName = experimentFile.fileName;
-
-        experiment = this.autoFillSomeMissingProperties(experiment);
-        experiment = this.deduceExperimentDetails(experiment);
 
         const newExperiment = plainToClass(ExperimentDTO, experiment);
-        let versionStatus = 0;
-        if (experiment.backendVersion) {
-          versionStatus = this.compareVersions(newExperiment.backendVersion, this.backendVersion);
-        }
 
         const experimentJSONValidationError = await this.validateExperimentJSON(newExperiment);
+        try {
+          const fileName = experimentFile.fileName;
+
+          experiment = this.autoFillSomeMissingProperties(experiment);
+          experiment = this.deduceExperimentDetails(experiment);
+
+          let versionStatus = 0;
+          if (experiment.backendVersion) {
+            versionStatus = this.compareVersions(newExperiment.backendVersion, this.backendVersion);
+          }
 
-        if (experimentJSONValidationError !== '') {
-          // If JSON is not valid, return error message
-          return { fileName: fileName, error: experimentJSONValidationError };
-        } else if (versionStatus !== 0) {
-          // If version is different, return appropriate warning message
-          const versionErrorMessage = versionStatus === 1 ? newVersionErrorMessage : oldVersionErrorMessage;
-          return { fileName: fileName, error: versionErrorMessage };
+          if (experimentJSONValidationError !== '') {
+            // If JSON is not valid, return error message
+            return { fileName: fileName, error: experimentJSONValidationError };
+          } else if (versionStatus !== 0) {
+            // If version is different, return appropriate warning message
+            const versionErrorMessage = versionStatus === 1 ? newVersionErrorMessage : oldVersionErrorMessage;
+            return { fileName: fileName, error: versionErrorMessage };
+          }
+          // If JSON is valid and version is the same, return null for no error
+          return { fileName: fileName, error: null };
+        } catch (error: any) {
+          return { fileName: experimentFile.fileName, error: experimentJSONValidationError };
```

Comment:
You can use the `fileName` variable you have defined instead of `experimentFile.fileName`